econml/dr/_drlearner.py

```diff
@@ -101,12 +103,18 @@ def predict(self, Y, T, X=None, W=None, *, sample_weight=None, groups=None):
         n = T.shape[0]
         Y_pred = np.zeros((T.shape[0], T.shape[1] + 1))
         T_counter = np.zeros(T.shape)
-        Y_pred[:, 0] = self._model_regression.predict(np.hstack([XW, T_counter])).reshape(n)
+        if self._binary_outcome and hasattr(self._model_regression, 'predict_proba'):
```

Comment:
Is it valid to have a regression model without a `predict_proba` method if `binary_outcome` is True?

Reply:
I was thinking it's not ideal but it should be fine (since this is the way it's been done all the way until now so maybe we should leave the option to use a regressor when y is binary). I will add a warning to communicate this is not the intended type of model to pass when discrete_outcome=True.